resources/default.rb

```diff
@@ -20,8 +20,8 @@
 
 actions :enable, :disable
 
-attribute :class_name, :kind_of => String, :name_attribute => true
-attribute :source, :default => nil, :kind_of => String
+attribute :class_name, :kind_of => String, :name_attribute => true, :required => true
+attribute :source, :kind_of => String, :required => true
```

Comment:
True.  There was a pull request to handle the nil case but it actually ended up causing a regression.  Trying to understand that is what led me to believe that the nil case never really made sense.